project-koku/koku
# [COST-4033] Use source application.extra.metered value for subs check

## Jira Ticket

[COST-4033](https://issues.redhat.com/browse/COST-4033)

## Description

This change will use the `application.extra.metered` field to determine if a source should process through the subs pipeline.

## Testing
Testing this requires sources running so you can either test in ephemeral or locally with sources spun up.

Local instructions:
1. Spin up sources on the same docker network as you will use for koku (if you need help with this, ask Corey)
2. Spin up koku with a subs worker but make sure you do not have `ENABLE_SUBS_DEBUG` set to True in your env. that should be false for this test. `make docker-up-min-with-subs`
3. Send a source to the sources api that includes `metered` set to `rhel` in the `applications` section:
```
"applications":[{
        "application_type_name":"cost-management",
        "extra":{"bucket":<my-bucket-name>, "metered":"rhel"},
        "source_name":<my-source-name>
    }]
```
4. See your source process all the way through and in the subs worker `docker-compose logs -f subs-worker` you should see the processing continue attempting to extract data and a log message that says `collecting subs report data`

## Notes

...
